modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveProcessor.java

```diff
@@ -55,52 +61,136 @@ public final class RemoveProcessor extends AbstractProcessor {
     public static final String TYPE = "remove";
 
     private final List<TemplateScript.Factory> fields;
+    private final List<String> fieldPatterns;
```

Comment:
@gaobinlong I believe this feature is adding confusion and complexity (as you may guess from this change set) by having more than one way to configure same things: `fields` and `fieldPatterns`, `excludeFields` and  `excludeFieldPatterns`, obviously bringing many questions, why not introducing `RemoveByPatternProcessor` (for example) instead?

Reply:
Too many parameters really confuse users, but `excludeFields` maybe reasonable kept in RemoveProcessor? If so I'll remove `fieldPatterns` and `excludeFieldPatterns` in this PR and open an issue and PR in the future.